File: 이코테/P149_음료수얼려먹기/P149.py


def func (y, x):
    graph[y][x] = 0
    for i in range(4):
        if graph[y + movey[i]][x+ movex[i]]:
            func(y + movey[i], x + movex[i])
            


n, m = map(int, input().split())
graph = []
movey = [0,1,0,-1]
movex = [1,0,-1,0]


graph.append( [0] * (m+2) )

for i in range(n):
    graph.append( [0] + list(map(lambda x : 1 - int(x)  , list(input())) ) + [0] )

graph.append( [0] * (m+2) )

result = 0
for i in range(1,n+2):
    for j in range(1, m+2):
        if graph[i][j]:
            result += 1
            func(i , j)
            
print(result)

File: 이코테/P149_음료수얼려먹기/P149.c
#include<stdio.h>

int map[1002][1002] = {0};
int n,m, result = 0;
char temp;
int movex[4] = {0,1,0,-1};
int movey[4] = {1,0,-1,0};
int func(int y, int x)
{
    map[y][x] = 0;
    for(int i=0; i < 4 ; i++)
    {
        if(map[y + movey[i]][x + movex[i]])
        {
            func(y + movey[i], x + movex[i]);
        }
    }
}

int main()
{
    scanf("%d %d",&n,&m);

    scanf("%c", &temp);
    for(int i=1 ; i <= n ; i++)
    {
        for(int j = 1 ; j <= m ; j++)
        {
            scanf("%c", &temp);
            map[i][j] = !(temp-'0');
        }
        scanf("%c", &temp);
    }



    for(int i=1 ; i <= n ; i++)
    {
        for(int j = 1 ; j <= m ; j++)
        {
            if(map[i][j])
            {
                func(i,j);
                result++;
            }
        }
    }

    printf("%d",result);
}